modules/internal/DefaultRectangular.chpl

```diff
@@ -809,6 +1013,241 @@ module DefaultRectangular {
+               minIndicesPerTask:int = dataParMinGranularity)
+      ref where !defRectSimpleDData {
+      if rank == 1 {
+        if !dom.stridable {
+          const first = getDataIndex(dom.dsiLow, getChunked=false);
+          const second = getDataIndex(dom.dsiLow+1, getChunked=false);
+          const step = (second-first);
+          if mdNumChunks == 1 {
+            const last = first + (dom.dsiNumIndices-1) * step;
+            const dd = theDataChunk(0);
+            for i in chpl_direct_pos_stride_range_iter(first, last, step) {
+              yield dd(i);
+            }
+          } else if dom.dsiNumIndices > 0 {
+            const (chunkLo, _) = getDataIndex(dom.dsiLow);
+            const (chunkHi, _) = getDataIndex(dom.dsiHigh);
+
+            // first chunk
+            {
+              const dd = theDataChunk(chunkLo);
+              const (_, hi) = mdChunk2Ind(chunkLo);
+              const (_, chunkFirst) = getDataIndex(dom.dsiLow);
+              const chunkLast = chunkFirst + (hi - dom.dsiLow) * step;
+              for i in chpl_direct_pos_stride_range_iter(chunkFirst,
+                                                         chunkLast,
+                                                         step) do
+                yield dd(i);
+            }
+
+            // middle chunks
+            for chunk in chunkLo+1..chunkHi-1 {
+              const dd = theDataChunk(chunk);
+              const (lo, hi) = mdChunk2Ind(chunk);
+              const (_, chunkFirst) = getDataIndex(lo);
+              const chunkLast = chunkFirst + (hi - lo) * step;
+              for i in chpl_direct_pos_stride_range_iter(chunkFirst,
+                                                         chunkLast,
+                                                         step) do
+                yield dd(i);
+            }
+
+            // last chunk
+            {
+              const dd = theDataChunk(chunkHi);
+              const (lo, _) = mdChunk2Ind(chunkHi);
+              const (_, chunkFirst) = getDataIndex(lo);
+              const chunkLast = chunkFirst + (dom.dsiHigh - lo) * step;
+              for i in chpl_direct_pos_stride_range_iter(chunkFirst,
+                                                         chunkLast,
+                                                         step) do
+                yield dd(i);
+            }
+          }
+        } else {
+          for i in dom do
+            yield dsiAccess(i);
+        }
+      } // if rank == 1 ...
+      else {
+        for i in dom do
+          yield dsiAccess(i);
+      }
+    }
+
+    iter these(param tag: iterKind,
+               tasksPerLocale = dataParTasksPerLocale,
+               ignoreRunning = dataParIgnoreRunningTasks,
+               minIndicesPerTask = dataParMinGranularity)
+      ref where tag == iterKind.standalone && !defRectSimpleDData {
+      if debugDefaultDist {
+        chpl_debug_writeln("*** In array standalone code");
```

Comment:
It might be good to make this message more specific and state that it's in the multi-ddata path. Though I suppose if you're debugging DefaultRectangular you probably know... oh well, consider it optional.